### ISSUE TITLE
Add support for GWF EnerCal F2 and similar devices

### ISSUE THREAD

**Issue by awor** [weetmuts/wmbusmeters#579]
I am trying to connect a GWF EnerCal F2 to your solution. This heat meter is popular in the German speaking part of Europe and is sold under numerous brands :
https://www.google.com/search?q=enercal+F2
Example: https://mgs-owl.de/mediafiles/Sonstiges/Deltamess/Montageanleitung/Bedien_10254_10255_10256_10257_10259_10260_10261_10262_10263_10264_10985.pdf

Could you please add support for this? Here is the debug output:

root@wmbus:/home/libmbus/binwmbusmeters --format=json --logtelegrams --debug --pollinterval=10 MAIN=/dev/ttyUSB0:mbus Tempo piigth:MAIN:2400 00309924 NOKEY
(wmbusmeters) version: 1.8.0-2-gd2918c5
(config) using device: MAIN=/dev/ttyUSB0:mbus
(config) number of meters: 1
(meter) created Tempo piigth encrypted
(lookup) with file/hex "/dev/ttyUSB0" mbus
(lookup) driver: mbus
(main) opening MAIN=/dev/ttyUSB0:mbus
Started config MAIN=mbus on /dev/ttyUSB0
(mbus) on /dev/ttyUSB0
(serial) before config: /dev/ttyUSB0 speed(B2400) input() output() control(CLOCAL|CREAD|CSIZE|PARENB) local() special_chars() modem(DTR|RTS)
(serial) after config: /dev/ttyUSB0 speed(B2400) input() output() control(CLOCAL|CREAD|CSIZE|PARENB) local() special_chars() modem(DTR|RTS)
(serialtty) opened /dev/ttyUSB0 fd 3 (mbus)
Sending NKE to mbus MAIN
(serial /dev/ttyUSB0) sent "1040004016"
(serial) EVENT thread interrupted
(main) regular reset of mbus on /dev/ttyUSB0 will happen every 82800 seconds
(wmbus) no alarm (expected activity) for mbus
(serial) registered regular callback HOT_PLUG_DETECTOR(0) every 2 seconds
(serial) waiting for stop
(meter) secondary addressing bus MAIN to address 00309924
(serial /dev/ttyUSB0) sent "680B0B6873FD5224993000FFFFFFFFAB16"
(serial) EVENT thread interrupted
(serial) received binary "E5"
(mbus) checkMBUSFrame
"E5"
(mbus) received E5 single byte frame.
(mbus) checkMBUSFrame
""
(mbus) less than 6 bytes, partial frame
(meter) polling Tempo 00309924 (secondary) with req ud2 bus MAIN
(serial /dev/ttyUSB0) sent "105BFD5816"
(serial) EVENT thread interrupted
(serial) received binary "68"
(mbus) checkMBUSFrame
"68"
(mbus) less than 6 bytes, partial frame
(serial) received binary "8E8E68"
(mbus) checkMBUSFrame
"688E8E68"
(mbus) less than 6 bytes, partial frame
(serial) received binary "08017224"
(mbus) checkMBUSFrame
"688E8E6808017224"
(mbus) not enough bytes, partial frame 8 148
(serial) received binary "993000"
(mbus) checkMBUSFrame
"688E8E6808017224993000"
(mbus) not enough bytes, partial frame 11 148
(serial) received binary "E61E08"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E08"
(mbus) not enough bytes, partial frame 14 148
(serial) received binary "040900"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E08040900"
(mbus) not enough bytes, partial frame 17 148
(serial) received binary "000004"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E08040900000004"
(mbus) not enough bytes, partial frame 20 148
(serial) received binary "068146"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E08040900000004068146"
(mbus) not enough bytes, partial frame 23 148
(serial) received binary "060004"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E08040900000004068146060004"
(mbus) not enough bytes, partial frame 26 148
(serial) received binary "1488F2"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2"
(mbus) not enough bytes, partial frame 29 148
(serial) received binary "350084"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084"
(mbus) not enough bytes, partial frame 32 148
(serial) received binary "401426"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426"
(mbus) not enough bytes, partial frame 35 148
(serial) received binary "E02600"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600"
(mbus) not enough bytes, partial frame 38 148
(serial) received binary "025B1F"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F"
(mbus) not enough bytes, partial frame 41 148
(serial) received binary "00025F"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F"
(mbus) not enough bytes, partial frame 44 148
(serial) received binary "1D0002"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002"
(mbus) not enough bytes, partial frame 47 148
(serial) received binary "621600"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600"
(mbus) not enough bytes, partial frame 50 148
(serial) received binary "04224988"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D000262160004224988"
(mbus) not enough bytes, partial frame 54 148
(serial) received binary "020004"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D000262160004224988020004"
(mbus) not enough bytes, partial frame 57 148
(serial) received binary "264988"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D000262160004224988020004264988"
(mbus) not enough bytes, partial frame 60 148
(serial) received binary "020004"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D000262160004224988020004264988020004"
(mbus) not enough bytes, partial frame 63 148
(serial) received binary "3B0000"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B0000"
(mbus) not enough bytes, partial frame 66 148
(serial) received binary "000004"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B0000000004"
(mbus) not enough bytes, partial frame 69 148
(serial) received binary "2C0000"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C0000"
(mbus) not enough bytes, partial frame 72 148
(serial) received binary "000004"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C0000000004"
(mbus) not enough bytes, partial frame 75 148
(serial) received binary "6D2510"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510"
(mbus) not enough bytes, partial frame 78 148
(serial) received binary "DE2784"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784"
(mbus) not enough bytes, partial frame 81 148
(serial) received binary "406E00"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E00"
(mbus) not enough bytes, partial frame 84 148
(serial) received binary "000000"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E00000000"
(mbus) not enough bytes, partial frame 87 148
(serial) received binary "848040"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E00000000848040"
(mbus) not enough bytes, partial frame 90 148
(serial) received binary "6E0000"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E0000"
(mbus) not enough bytes, partial frame 93 148
(serial) received binary "0000C4"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4"
(mbus) not enough bytes, partial frame 96 148
(serial) received binary "800006"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006"
(mbus) not enough bytes, partial frame 99 148
(serial) received binary "B1450600"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600"
(mbus) not enough bytes, partial frame 103 148
(serial) received binary "C48000"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000"
(mbus) not enough bytes, partial frame 106 148
(serial) received binary "143BE4"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE4"
(mbus) not enough bytes, partial frame 109 148
(serial) received binary "3500C4"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4"
(mbus) not enough bytes, partial frame 112 148
(serial) received binary "C00014"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C00014"
(mbus) not enough bytes, partial frame 115 148
(serial) received binary "2FD226"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD226"
(mbus) not enough bytes, partial frame 118 148
(serial) received binary "00C4C0"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0"
(mbus) not enough bytes, partial frame 121 148
(serial) received binary "006E00"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00"
(mbus) not enough bytes, partial frame 124 148
(serial) received binary "000000"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000"
(mbus) not enough bytes, partial frame 127 148
(serial) received binary "C48040"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000C48040"
(mbus) not enough bytes, partial frame 130 148
(serial) received binary "6E0000"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000C480406E0000"
(mbus) not enough bytes, partial frame 133 148
(serial) received binary "0000C2"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000C480406E00000000C2"
(mbus) not enough bytes, partial frame 136 148
(serial) received binary "80006C"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000C480406E00000000C280006C"
(mbus) not enough bytes, partial frame 139 148
(serial) received binary "C1271F"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000C480406E00000000C280006CC1271F"
(mbus) not enough bytes, partial frame 142 148
(serial) received binary "000000"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000C480406E00000000C280006CC1271F000000"
(mbus) not enough bytes, partial frame 145 148
(serial) received binary "00D516"
(mbus) checkMBUSFrame
"688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000C480406E00000000C280006CC1271F00000000D516"
(mbus) received full frame.
(mbus) parse MBUS DLL @0 148
(mbus) "688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000C480406E00000000C280006CC1271F00000000D516"
(dvparser) cannot handle dif 1F ignoring rest of telegram.
(dvparser) found new format "04060414844014025B025F026204220426043B042C046D84406E8480406EC4800006C4800014C4C00014C4C0006EC480406EC280006C" with hash 620f, remembering!
(meter) Tempo: for me? 01,00309924 in 00309924
(meter) Tempo: meter detection did not match the selected driver unknown! correct driver is: unknown!
(meter) Not printing this warning again for id: 00000000 mfct: (@@@) Unknown (0x00) type: Other (0x00) ver: 0x00
(meter) please consider opening an issue at https://github.com/weetmuts/wmbusmeters/
(meter) to add support for this unknown mfct,media,version combination
(meter) Tempo: yes for me
(meter) Tempo(1) piigth handling telegram from 00309924
(meter) Tempo 00309924 "688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000C480406E00000000C280006CC1271F00000000D516"
(mbus) parse MBUS DLL @0 148
(mbus) "688E8E6808017224993000E61E080409000000040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000C480406E00000000C280006CC1271F00000000D516"
(dvparser) cannot handle dif 1F ignoring rest of telegram.
telegram=|688E8E6808017224993000E61E080409000000_040681460600041488F2350084401426E02600025B1F00025F1D0002621600042249880200042649880200043B00000000042C00000000046D2510DE2784406E000000008480406E00000000C4800006B1450600C48000143BE43500C4C000142FD22600C4C0006E00000000C480406E00000000C280006CC1271F00000000D516|+4
(meters) using field info on_time(Time)[3] to extract 0422
(meters) using field info operating_time(Time)[2] to extract 0426
(meters) using field info meter_datetime(Text)[4] to extract 046D
(piigth) 000 : 68 start
(piigth) 001 : 8e length (142 bytes)
(piigth) 002 : 8e length again (142 bytes)
(piigth) 003 : 68 start
(piigth) 004 : 08 dll-c (RSP_UD2)
(piigth) 005 : 01 dll-a primary (1)
(piigth) 006 : 72 tpl-ci (long header)
(piigth) 007 : 24993000 tpl-id (00309924)
(piigth) 011 : e61e tpl-mfct (GWF)
(piigth) 013 : 08 tpl-version
(piigth) 014 : 04 tpl-type (Heat meter)
(piigth) 015 : 09 tpl-acc-field
(piigth) 016 : 00 tpl-sts-field (OK)
(piigth) 017 : 0000 tpl-cfg 0000 ( )
(piigth) 019 : 04 dif (32 Bit Integer/Binary Instantaneous value)
(piigth) 020 : 06 vif (Energy kWh)
(piigth) 021 C?: 81460600
(piigth) 025 : 04 dif (32 Bit Integer/Binary Instantaneous value)
(piigth) 026 : 14 vif (Volume 10⁻² m³)
(piigth) 027 C?: 88F23500
(piigth) 031 : 84 dif (32 Bit Integer/Binary Instantaneous value)
(piigth) 032 : 40 dife (subunit=1 tariff=0 storagenr=0)
(piigth) 033 : 14 vif (Volume 10⁻² m³)
(piigth) 034 C?: 26E02600
(piigth) 038 : 02 dif (16 Bit Integer/Binary Instantaneous value)
(piigth) 039 : 5B vif (Flow temperature °C)
(piigth) 040 C?: 1F00
(piigth) 042 : 02 dif (16 Bit Integer/Binary Instantaneous value)
(piigth) 043 : 5F vif (Return temperature °C)
(piigth) 044 C?: 1D00
(piigth) 046 : 02 dif (16 Bit Integer/Binary Instantaneous value)
(piigth) 047 : 62 vif (Temperature difference 10⁻¹ K/°C)
(piigth) 048 C?: 1600
(piigth) 050 : 04 dif (32 Bit Integer/Binary Instantaneous value)
(piigth) 051 : 22 vif (On time hours)
(piigth) 052 C!: 49880200 ("on_time_h":46.100278)
(piigth) 056 : 04 dif (32 Bit Integer/Binary Instantaneous value)
(piigth) 057 : 26 vif (Operating time hours)
(piigth) 058 C!: 49880200 ("operating_time_h":46.100278)
(piigth) 062 : 04 dif (32 Bit Integer/Binary Instantaneous value)
(piigth) 063 : 3B vif (Volume flow l/h)
(piigth) 064 C?: 00000000
(piigth) 068 : 04 dif (32 Bit Integer/Binary Instantaneous value)
(piigth) 069 : 2C vif (Power 10¹ W)
(piigth) 070 C?: 00000000
(piigth) 074 : 04 dif (32 Bit Integer/Binary Instantaneous value)
(piigth) 075 : 6D vif (Date and time type)
(piigth) 076 C!: 2510DE27 ("meter_datetime":"2022-07-30 16:37")
(piigth) 080 : 84 dif (32 Bit Integer/Binary Instantaneous value)
(piigth) 081 : 40 dife (subunit=1 tariff=0 storagenr=0)
(piigth) 082 : 6E vif (Units for H.C.A.)
(piigth) 083 C?: 00000000
(piigth) 087 : 84 dif (32 Bit Integer/Binary Instantaneous value)
(piigth) 088 : 80 dife (subunit=0 tariff=0 storagenr=0)
(piigth) 089 : 40 dife (subunit=2 tariff=0 storagenr=0)
(piigth) 090 : 6E vif (Units for H.C.A.)
(piigth) 091 C?: 00000000
(piigth) 095 : C4 dif (32 Bit Integer/Binary Instantaneous value storagenr=1)
(piigth) 096 : 80 dife (subunit=0 tariff=0 storagenr=1)
(piigth) 097 : 00 dife (subunit=0 tariff=0 storagenr=1)
(piigth) 098 : 06 vif (Energy kWh)
(piigth) 099 C?: B1450600
(piigth) 103 : C4 dif (32 Bit Integer/Binary Instantaneous value storagenr=1)
(piigth) 104 : 80 dife (subunit=0 tariff=0 storagenr=1)
(piigth) 105 : 00 dife (subunit=0 tariff=0 storagenr=1)
(piigth) 106 : 14 vif (Volume 10⁻² m³)
(piigth) 107 C?: 3BE43500
(piigth) 111 : C4 dif (32 Bit Integer/Binary Instantaneous value storagenr=1)
(piigth) 112 : C0 dife (subunit=1 tariff=0 storagenr=1)
(piigth) 113 : 00 dife (subunit=1 tariff=0 storagenr=1)
(piigth) 114 : 14 vif (Volume 10⁻² m³)
(piigth) 115 C?: 2FD22600
(piigth) 119 : C4 dif (32 Bit Integer/Binary Instantaneous value storagenr=1)
(piigth) 120 : C0 dife (subunit=1 tariff=0 storagenr=1)
(piigth) 121 : 00 dife (subunit=1 tariff=0 storagenr=1)
(piigth) 122 : 6E vif (Units for H.C.A.)
(piigth) 123 C?: 00000000
(piigth) 127 : C4 dif (32 Bit Integer/Binary Instantaneous value storagenr=1)
(piigth) 128 : 80 dife (subunit=0 tariff=0 storagenr=1)
(piigth) 129 : 40 dife (subunit=2 tariff=0 storagenr=1)
(piigth) 130 : 6E vif (Units for H.C.A.)
(piigth) 131 C?: 00000000
(piigth) 135 : C2 dif (16 Bit Integer/Binary Instantaneous value storagenr=1)
(piigth) 136 : 80 dife (subunit=0 tariff=0 storagenr=1)
(piigth) 137 : 00 dife (subunit=0 tariff=0 storagenr=1)
(piigth) 138 : 6C vif (Date type G)
(piigth) 139 C?: C127
(meters) render field operating_time(Time)[2] with dventry @58 key 0426 data 49880200
(meters) "operating_time_h":46.100278
(meters) render field on_time(Time)[3] with dventry @52 key 0422 data 49880200
(meters) "on_time_h":46.100278
(meters) render field meter_datetime(Text)[4] with dventry @76 key 046D data 2510DE27
(meters) "meter_datetime":"2022-07-30 16:37"
{"media":"heat","meter":"piigth","name":"Tempo","id":"00309924","operating_time_h":46.100278,"on_time_h":46.100278,"meter_datetime":"2022-07-30 16:37","temperature_c":null,"average_temperature_1h_c":null,"average_temperature_24h_c":null,"relative_humidity_rh":null,"relative_humidity_1h_rh":null,"relative_humidity_24h_rh":null,"timestamp":"2022-07-30T11:02:02Z","device":"MAIN","rssi_dbm":0}
(mbus) checkMBUSFrame
""
(mbus) less than 6 bytes, partial frame
^C(serial) stopping manager
(serial) MAIN thread interrupted
(serial) closing 1 devices
(wmbus) disconnected /dev/ttyUSB0 mbus
(serialtty) closed /dev/ttyUSB0 (mbus)
(serial) TIMER thread interrupted
(serial) event loop stopped!
(wmbus) deleted mbus
(serial) closing 1 devices
root@wmbus:/home/libmbus/bin#

**Comment by BIBOLV** [comment]
Sensostar driver can decode some values, could you verify that those are correct?
```
{
    "media":"heat",
    "meter":"sensostar",
    "name":"",
    "id":"00309924",
    "meter_timestamp":"2022-07-30 16:37",
    "total_kwh":411265,
    "power_kw":0,
    "flow_water_m3h":0,
    "forward_c":31,
    "return_c":29,
    "difference_c":2.2,
    "total_water_m3":35354.96,
    "current_status":null,
    "timestamp":"2022-08-05T20:54:34Z"
}
```

**Comment by awor** [comment]
The json data you posted looks good, many thanks for your input @BIBOLV !

I tried this on my side, but it isn't returning any json in the console: How must I proceed to test on my end?

root@wmbus:~# wmbusmeters --format=json --logtelegrams --debug  --pollinterval=10 MAIN=/dev/ttyUSB0:mbus Tempo sensostar:MAIN:2400 00309924 NOKEY
(wmbusmeters) version: 1.8.0-2-gd2918c5
(config) using device: MAIN=/dev/ttyUSB0:mbus
(config) number of meters: 1
(lookup) with file/hex "/dev/ttyUSB0" mbus
(lookup) driver: mbus
(main) opening MAIN=/dev/ttyUSB0:mbus
Started config MAIN=mbus on /dev/ttyUSB0
(mbus) on /dev/ttyUSB0
(serial) before config: /dev/ttyUSB0 speed(B2400) input() output() control(CLOCAL|CREAD|CSIZE|PARENB) local() special_chars() modem(DTR|RTS)
(serial) after config:  /dev/ttyUSB0 speed(B2400) input() output() control(CLOCAL|CREAD|CSIZE|PARENB) local() special_chars() modem(DTR|RTS)
(serialtty) opened /dev/ttyUSB0 fd 3 (mbus)
Sending NKE to mbus MAIN
(serial /dev/ttyUSB0) sent "1040004016"
(serial) EVENT thread interrupted
(main) regular reset of mbus  on /dev/ttyUSB0 will happen every 82800 seconds
(wmbus) no alarm (expected activity) for mbus
(serial) registered regular callback HOT_PLUG_DETECTOR(0) every 2 seconds
(serial) waiting for stop

**Comment by awor** [comment]
Ah, this could be the problem: `(serial) internal error after select! errno=Illegal seek`
Hi @weetmuts, would it be possible for you to look into this error message? Maybe this is why polling is not working when using config files (command line works, see above):

root@wmbus:~# wmbusmeters --useconfig=/etc --debug
(config) "device" "MAIN=/dev/ttyUSB0:mbus:2400"
(config) "logtelegrams" "false"
(config) "format" "json"
(config) "meterfiles" "/var/lib/wmbusmeters/meter_readings"
(config) "meterfilesaction" "overwrite"
(config) "#logfile=/var/log/wmbusmeters/wmbusmeters.log" ""
(config) "logfile" "/dev/stdout"
(config) "shell" "/usr/bin/mosquitto_pub -h 172.22.1.151 -t wmbusmeters/"$METER_ID" -m "$METER_JSON""
(config) "" ""
(config) loading meter file /etc/wmbusmeters.d/EnerCal
(config) name=EnerCal
(config) driver=sensostar:mbus
(config) pollinterval=10s
(config) id=00309924
(config) overriding loglevel with debug
(wmbusmeters) using log file /dev/stdout
(wmbusmeters) version: 1.8.0-2-gd2918c5
(config) store meter files in: "/var/lib/wmbusmeters/meter_readings"
(config) using device: MAIN=/dev/ttyUSB0:mbus:2400
(config) number of meters: 1
(lookup) with file/hex "/dev/ttyUSB0" mbus
(lookup) driver: mbus
(main) opening MAIN=/dev/ttyUSB0:mbus:2400
Started config MAIN=mbus on /dev/ttyUSB0
(mbus) on /dev/ttyUSB0
(serial) before config: /dev/ttyUSB0 speed(B2400) input() output() control(CLOCAL|CREAD|CSIZE|PARENB) local() special_chars() modem(DTR|RTS)
(serial) after config:  /dev/ttyUSB0 speed(B2400) input() output() control(CLOCAL|CREAD|CSIZE|PARENB) local() special_chars() modem(DTR|RTS)
(serialtty) opened /dev/ttyUSB0 fd 3 (mbus)
Sending NKE to mbus MAIN
(serial /dev/ttyUSB0) sent "1040004016"
(serial) EVENT thread interrupted
**`(serial) internal error after select! errno=Illegal seek`**
(main) regular reset of mbus  on /dev/ttyUSB0 will happen every 82800 seconds
(wmbus) no alarm (expected activity) for mbus
(serial) registered regular callback HOT_PLUG_DETECTOR(0) every 2 seconds
(serial) waiting for stop 

Many thanks in advance!

**Comment by weetmuts** [comment]
Hmm, wmbusmeters never seek in the serial device. So is this a problem with the serial driver/device?

Which is the mbus device that you have connected to your /dev/ttyUSB0 ?

Can you also post the dmesg output when you plug-in the device?

**Comment by weetmuts** [comment]
I have pushed a new driver enercal. You can check the decode here:
https://wmbusmeters.org/analyze/688e8e6808017224993000e61e080406000000040681460600041488f2350084401426e02600025B2600025f220002622a00042247880200042647880200043B00000000042c00000000046d2d0ede2784406e000000008480406e00000000c4800006B1450600c48000143Be43500c4c000142fd22600c4c0006e00000000c480406e00000000c280006cc1271f00000000f416

**Comment by awor** [comment]
Many thanks for looking into this! At first glance, your new enercal driver is looking very good :) It's now returning the values/json where it previously failed. I still have some mqtt connection issue I need to investigate tomorrow. I will get back to you shortly. Until then, tack så mycket!

**Comment by weetmuts** [comment]
Great! What is the name of the mbus master device that you use? I.e. the box that translates the serial data to mbus line levels?

Do you know why the  enercal has HCA values? And what are the subunits in the device used for?

**Comment by awor** [comment]
I am using the same device that [atze09](https://github.com/atze09) is using in https://github.com/weetmuts/wmbusmeters/issues/435 which is this one:
https://www.aliexpress.com/item/4000787535232.html

Sorry, I am not familiar with HCA values and haven't got a clue what the subunits are used for. You can see what is available on the display of the unit by looking at pages 7-8 in the document listed here https://github.com/weetmuts/wmbusmeters/issues/579#issue-1330233871. Maybe it's one of the impulse registers? Once I connect wmbusmeters with my Home Assistant and get more familiar with the data, maybe I can give a better answer.

**Comment by awor** [comment]
Okay, there seems to be a problem with the json payload, it contains a CR LF right after the meter name:
![image](https://user-images.githubusercontent.com/3464535/184427751-5cf75ad3-1aa5-4755-a3f9-dc291569ad66.png)
this is not valid and breaks the payload, hence it cannot be properly processed. I don't see anything I can do to cause or influence this. Here is my meter config in an editor:
![image](https://user-images.githubusercontent.com/3464535/184427978-1a8066c2-992e-41fe-b2f6-2a050997b640.png)
when you have a moment, could you please look into this?
Generally, the parsing of the meter config seems to be a little flaky: Putting the parameters in a different order doesn't appear to work, also adding a CR LF at the end of the last line will break the config.

**Comment by awor** [comment]
I had created my meter config file in WinSCP and Notepad++. I finally realized that Unix EOL is not CRLF but just LF. This caused all the havoc I described above. In NP++ I now set Unix (LF) in Edit > EOL Conversion, saved the file, and now all works as expected. It took me some time to figure this out. @weetmuts you might want to consider adding this warning in the documentation, i.e. that the config files will break if Windows style CRLF is used instead of Unix style LF line endings. Again, many thanks for your great work. A number of people are probably using your solution to optimize the energy used in their homes (like me). In other words, your great project is a contribution to saving the planet :-)
